internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -210,17 +212,26 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, s *scope) (ctr
 	}
 	s.infraCluster = obj
 
-	// Determine if the infrastructure provider is ready.
-	ready, err := external.IsReady(s.infraCluster)
+	// Determine contract version used by the InfrastructureCluster.
+	contractVersion, err := utilconversion.GetContractVersion(ctx, r.Client, obj.GroupVersionKind())
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if ready && !cluster.Status.InfrastructureReady {
-		log.Info("Infrastructure provider has completed cluster infrastructure provisioning and reports status.ready", cluster.Spec.InfrastructureRef.Kind, klog.KObj(s.infraCluster))
+
+	// Determine if the InfrastructureCluster is provisioned.
+	provisioned, err := contract.InfrastructureCluster().Provisioned(contractVersion).Get(obj)
```

Comment:
Let's stay consistent and use `s.infraCluster` instead of `obj`:

```suggestion
	provisioned, err := contract.InfrastructureCluster().Provisioned(contractVersion).Get(s.infraCluster)
```